File: strategy-pattern/py/concrete_strategies.py
from strategy_interface import StrategyInterface

class BardStrategy(StrategyInterface):

    def __init__(self) -> None:
        super().__init__()

    def generateResponse(self):
        print("This is bard implementation of generateResponse")

class GeminiStrategy(StrategyInterface):

    def __init__(self) -> None:
        super().__init__()

    def generateResponse(self):
        print("This is Gemini implementation of generateResponse")

class ChatgptStrategy(StrategyInterface):

    def __init__(self) -> None:
        super().__init__()

    def generateResponse(self):
        print("This is ChatGPT implementation of generateResponse")

File: strategy-pattern/ts/concreteStrategies.ts
import AIStrategyInterface from "./strategyInterface";

export class BardStrategy extends AIStrategyInterface{

    generateResponse(){
        console.log("I am bard, and this is my implementation of generateresponse")
    }
}

export class GeminiStrategy extends AIStrategyInterface{
    generateResponse(){
        console.log("I am Gemini, and this is my implementation of generateResponse")
    }
}

export class ChatGPTStrategy extends AIStrategyInterface{
    generateResponse(){
        console.log("I am ChatGPT, and this is my implementatiion of generateResponse")
    }
}

